---
title: "Stone Age Plague"
output: html_notebook
---

##Description Notebook
This notebook will contain any bits of analysis conducted in samples positive for plague during the Stone Age (and possibly later, only prehistorical genomes). 

#Checking branch OOH003-RISE505-ARS007-KZL002
```{r}
library(tidyverse)
snpTable_OOH003c <- read.delim("Data/2020-07-09_LNBA_leprosy_enterica_comp/LNBA_transect/genotyped_snpTable_OOH003_c.tsv")
noncalls <- c(".","N")

  snpTable_OOH003c %>%
  select(Position, OOH003_c, RISE505, ARS007, KZL002, KLE048) %>%
  filter(!OOH003_c %in% noncalls & RISE505 %in% noncalls & KLE048 %in% noncalls) %>%
  filter(OOH003_c == ARS007 | OOH003_c == KZL002)

snpTable_OOH003c %>%
  select(Position, OOH003_c, RISE505, ARS007, KZL002, KLE048) %>%
  filter(!RISE505 %in% noncalls & OOH003_c %in% noncalls & KLE048 %in% noncalls) %>%
  filter(RISE505 == ARS007 | OOH003_c == KZL002) 

snpTable_OOH003c %>%
  select(Position, OOH003_c, RISE505, ARS007, KZL002, KLE048) %>%
  filter(!RISE505 %in% noncalls & !OOH003_c %in% noncalls & !ARS007 %in% noncalls & !KZL002 %in% noncalls & KLE048 %in% noncalls)

snpTable_OOH003c %>%
  select(Position, OOH003_c, RISE505, ARS007, KZL002, KLE048) %>%
  filter(!RISE505 %in% noncalls & !OOH003_c %in% noncalls & ARS007 %in% noncalls & KZL002 %in% noncalls & KLE048 %in% noncalls)

##Branch RK1 and RISE509
snpTable_OOH003c %>%
  select(Position, RISE509, RK1001.C, GEN72, GRS004, Gyvakarai1, XXX001.A18962.69.72, GZL002.A0101_02.YP2.1, GZL001.A0101_02.YP2.1, KunilaII, HUT004.A_1343UnTal85, HOP001, HOP004, PST006.A_Post6, KLE031, KLE048, RISE505, OOH003_c, ARS007, KZL002) %>%
  filter(!RISE509 %in% noncalls & RK1001.C %in% noncalls & !GEN72 %in% noncalls)

snpTable_OOH003c %>%
  filter(HUT004.A_1343UnTal85 == KLZ001 & HUT004.A_1343UnTal85 != HOP001)
```
#Virulence analysis
###Chromosome
```{r}
library(ggplot2)
library(readr)
library(viridis)
library(ggpubr)

chromosome_heatmap <- read_tsv(file = "Data/Virulence/Chromosome/all_chromosome_reviews.tsv",
                               col_names = c("genome", "perc", "genes")) %>%
  mutate(genome= ifelse(genome=="1343UNTA85", "1343UNTAL85", genome))

chromosome_heatmap_reduced <- chromosome_heatmap[!grepl("filamentous_prophage[1-9]", chromosome_heatmap$genes),]


genomes_ordered <- read_csv(file = "Data/Virulence/genome_names_reviews.csv",
                            col_names = c("Order")) %>%
  mutate(Order= ifelse(Order=="1343UNTA85", "1343UNTAL85", Order))

y_order <- as.vector(genomes_ordered$Order)
chromosome_heatmap_reduced$genome <- factor(chromosome_heatmap_reduced$genome, levels=y_order)
x_order <- as.vector(unique(chromosome_heatmap_reduced$genes))
chromosome_heatmap_reduced$genes <- factor(chromosome_heatmap_reduced$genes, levels = x_order)


chromosome_plot <- ggplot(data = chromosome_heatmap_reduced, mapping = aes(x = genes, y = genome, fill = perc)) +
  geom_raster() +
  scale_fill_viridis_c(option = "inferno") +
  labs(fill = "Percentage", title = "Chromosome", x = NULL) +
  theme(axis.text.x = element_text(size = 12, angle = 90, hjust = 1, vjust = 0.5),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 16),
        axis.title.y = element_blank(),
        plot.title = element_text(hjust = 0.5),
        legend.position = "none")
```
###Plasmids
```{r}
plasmids_heatmap <- read_tsv(file = "Data/Virulence/Plasmids/all_plasmids_reviews.tsv",
                             col_names = c("genome", "perc", "genes", "plasmid")) %>%
  mutate(genome= ifelse(genome=="1343UNTA85", "1343UNTAL85", genome))


plasmids_heatmap$genome <- factor(plasmids_heatmap$genome, levels=y_order)
x_order_plasmids <- as.vector(unique(plasmids_heatmap$genes))
plasmids_heatmap$genes <- factor(plasmids_heatmap$genes, levels = x_order_plasmids)

plasmids_plot <- ggplot(data = plasmids_heatmap, mapping = aes(x = genes, y = genome, fill = perc)) +
  geom_raster() +
  scale_fill_viridis_c(option = "inferno")+
  facet_grid(~ plasmid, scales = "free_x", space = "free_x") +
  theme(axis.text.x = element_text(size = 12, angle = 90, hjust = 1, vjust = 0.5),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 16),
        axis.title.y = element_blank(),
        strip.text.x = element_text(size = 12),
        strip.background = element_rect(fill = "transparent"),
        plot.margin = margin(0,200,0,200),
        panel.spacing = unit(4, "lines")
        ) +
  labs(fill = "Percentage", x = NULL)

Figure2 <- ggarrange(chromosome_plot, plasmids_plot, nrow = 2, heights = c(1,0.8))
ggsave("Figure3.pdf", plot = Figure2, device = cairo_pdf, path = "Plots/", height = 35, width = 45, units = "cm", limitsize = FALSE)
ggsave("Figure3.png", plot = Figure2, device = "png", path = "Plots/", height = 35, width = 45, units = "cm", limitsize = FALSE)

```

#Genetic vs geographical distance
###Package and function loading
```{r}
#Calculate pairwise genetic distance
library(tidyverse)
library(geosphere)
library(reshape2)
library(ggplot2)
library(ape)
library(plotly)
library(ggpubr)
library(vegan)
###FUNCTIONS###

matrixToLong <- function(matrix, name, levels){
  longdf <- melt(matrix, value.name = name)
  longdf$Var1 <- factor(longdf$Var1, levels = levels)
  longdf$Var2 <- factor(longdf$Var2, levels = levels)
  return(longdf)
}

GenDistanceCalculator <- function(snpData, model, pairwise, levels, snpTable){
  percentDistance <- dist.dna(snpData, model = model, 
                              as.matrix = TRUE, 
                              pairwise.deletion = pairwise)
#https://rdrr.io/cran/ape/man/dist.gene.html
  rownames(percentDistance) <- unique(snpTable$Genomes)
  colnames(percentDistance) <- unique(snpTable$Genomes)
  return(percentDistance)
}

GeoDistanceCalculator <- function(metadata, levels) {
  metadataFiltered <- metadata %>%
  filter(Name %in% levels)
  
  lonlatmatrix <- metadataFiltered %>%
    select(Longitud, Latitude) %>%
    data.matrix()
  distanceMatrix <- distm(lonlatmatrix, fun = distCosine)
  rownames(distanceMatrix) <- metadataFiltered$Name
  colnames(distanceMatrix) <- metadataFiltered$Name
  return(distanceMatrix)
}

C14DistanceCalculator <- function(metadata, levels) {
  metadataFiltered <- metadata %>%
  filter(Name %in% levels)
  Median.C14 <- metadataFiltered$Median.C14
  medianDateDiff <- outer(Median.C14, Median.C14, '-')
  medianDateDiff <- abs(medianDateDiff)
  rownames(medianDateDiff) <- metadataFiltered$Name
  colnames(medianDateDiff) <- metadataFiltered$Name
  return(medianDateDiff)
}

matrixPlotting <- function(df, fi, title) {
  ggplot(df, aes(x= Var2, y=Var1))+
  geom_raster(aes(fill=fi)) +
  scale_fill_viridis_c(option="plasma") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x="Samples", y="Samples", title=title)
}

plotGenTime <- function(DNADist, GeoDist, TimeDist, mantelResults) {
  GenGeoTimeD <- DNADist %>%
    left_join(GeoDist) %>%
    left_join(TimeDist) 
  
  GenTimeCorr <- ggplot(GenGeoTimeD, aes(x= GenD, y=C14_distance, color=GeoD)) +
    geom_point() +
    #geom_smooth(method = lm, se = FALSE) +
    #geom_smooth() +
    scale_color_viridis_c(option = "plasma") +
    labs(title = paste("Mantel Statistic r = ", round(mantelResults$statistic, digits = 4), ", p-value = ", round(mantelResults$signif, digits = 4), sep = ""), x = "Genetic distance", y= "Temporal distance", colour= "Geographical \n distance") +
    theme_bw()
  return(GenTimeCorr)
}

plotGenGeo <- function(DNADist, GeoDist, TimeDist, mantelResults) {
  GenGeoTimeD <- DNADist %>%
    left_join(GeoDist) %>%
    left_join(TimeDist) %>%
    filter(C14_distance > 0)
  
  GenGeoCorr <- ggplot(GenGeoTimeD, aes(x= GenD, y=GeoD, color=C14_distance)) +
    geom_point() +
   # geom_smooth(method = lm, se = FALSE) +
#    geom_smooth() +
    scale_color_viridis_c(option = "plasma") +
    labs(title = paste("Mantel Statistic r = ", round(mantelResults$statistic, digits = 4), ", p-value = ", round(mantelResults$signif, digits = 4), sep = ""), x = "Genetic distance", y= "Geographical distance", colour= "Temporal \n distance") +
    theme_bw()
  return(GenGeoCorr)
}


plotGenTimelm <- function(DNADist, GeoDist, TimeDist) {
  GenGeoTimeD <- DNADist %>%
  left_join(GeoDist) %>%
  left_join(TimeDist) %>%
  filter(C14_distance > 0)

lmGenTime <- lm(GenD ~ C14_distance, data = GenGeoTimeD)
GenTimeCorr <- ggplot(GenGeoTimeD, aes(x= GenD, y=C14_distance, color=GeoD)) +
  geom_point() +
  geom_smooth(method = lm) +
  scale_color_viridis_c(option = "plasma") +
  labs(title = paste("Adj R2 =", round(summary(lmGenTime)$adj.r.squared, digits = 4), ",p-value =",  summary(lmGenTime)$coef[2,4]), x = "Genetic distance", y= "Temporal distance", colour= "Geographical \n distance") +
  theme_bw()
return(GenTimeCorr)
}

plotGenGeolm <- function(DNADist, GeoDist, TimeDist) {
  GenGeoTimeD <- DNADist %>%
  left_join(GeoDist) %>%
  left_join(TimeDist) %>%
  filter(C14_distance > 0)

lmGenGeo <- lm(GenD ~ GeoD, data = GenGeoTimeD)
GenGeoCorr <- ggplot(GenGeoTimeD, aes(x= GenD, y=GeoD, color=C14_distance)) +
  geom_point() +
  geom_smooth(method = lm) +
  scale_color_viridis_c(option = "plasma") +
  labs(title = paste("Adj R2 =", round(summary(lmGenGeo)$adj.r.squared, digits = 4), ",p-value =", summary(lmGenGeo)$coef[2,4]), x = "Genetic distance", y= "Geographical distance", colour= "Temporal \n distance") +
  theme_bw()
return(GenGeoCorr)
}

###END FUNCTIONS ###
```
###LNBA branch
```{r}
##Reading metadata file
metadata <- read.csv("Data/2020-07-09_LNBA_leprosy_enterica_comp/LNBA_transect/Metadata_coordinates_dating_sex_updated_def.csv") %>%
  filter(!Name=="XXX001.A18956-61")

##Reading snpTable and selecting genomes present in the metadata file
snpTable <- read.delim("Data/2020-07-09_LNBA_leprosy_enterica_comp/LNBA_transect/LNBAdataset2021_gok2_RV2039_SNPExcluded_genotyped.tsv", check.names = FALSE) %>%
  pivot_longer(c(-Position, -Ref), names_to = "Genomes", values_to = "Call") %>%
  filter(Genomes %in% metadata$Name) %>%
  mutate(CallDef = ifelse(Call == ".", as.character(Ref), as.character(Call))) %>%
  select(-Ref, -Call) %>%
    filter(!Genomes=="XXX001.A18956-61")

#missingData <- snpTable %>%
#  filter(CallDef == "N") %>%
#  group_by(Position) %>%
#  summarise(percent=(n()/length(unique(snpTable$Genomes)))*100)

GenomesPresent <- unique(snpTable$Genomes) 
#Prepare data for calculating the genetic distance and excluding positions where all the genomes have the same call
snpDataPestis <- snpTable %>%
  pivot_wider(names_from = Position, values_from = CallDef) %>%
  select(where(~ n_distinct(.) > 1)) %>%
  as.matrix() %>%
  as.DNAbin()

levelsPhylogeny <- metadata$Name

#Calculate genetic distance matrix and converting it to long for plotting
DistDNA <- GenDistanceCalculator(snpDataPestis, "raw", TRUE, levelsPhylogeny, snpTable)
DNAlong <- matrixToLong(DistDNA, "GenD", levelsPhylogeny)


#Calculate geographic distance matrix and converting it to long for plotting
GeoDist <- GeoDistanceCalculator(metadata, levelsPhylogeny)
Geolong <- matrixToLong(GeoDist, "GeoD", levelsPhylogeny)

#Difference median C14 and converting it to long for plotting
C14Dist <- C14DistanceCalculator(metadata, levelsPhylogeny)
C14long <- matrixToLong(C14Dist, "C14_distance", levelsPhylogeny)

#Ordering the genetic matrix to have the same order for calculating Mantel's stats
DistDNAGeo <- DistDNA[rownames(GeoDist),rownames(GeoDist)]
DistDNATime <- DistDNA[rownames(C14Dist),rownames(C14Dist)]

mantelGenGeo <- mantel(DistDNAGeo, GeoDist, method = "spearman", permutations = 9999)
mantelGenTime <- mantel(DistDNATime, C14Dist, method = "spearman", permutations = 9999)

GenTimeCorr <- plotGenTime(DNAlong, Geolong, C14long, mantelGenTime)
GenGeoCorr <- plotGenGeo(DNAlong, Geolong, C14long, mantelGenGeo)

plotLNBA <- ggarrange(GenTimeCorr, GenGeoCorr, ncol = 2)
plotLNBAannotated <- annotate_figure(plotLNBA, top= text_grob("Y. pestis LNBA branch", face = "italic"))

##Convert to function
#plot_ly(data = GenGeoTimeD, x= ~GenD, y= ~C14_distance, z= ~GeoD, type = "scatter3d", mode="markers", color = ~Var2)
```
###Second Pandemic Branch
```{r}
##Reading metadata file
metadata <- read.csv("Data/2020-07-09_LNBA_leprosy_enterica_comp/2ndPandemic_transect/Metadata_2ndpandemic.csv")

##Reading snpTable and selecting genomes present in the metadata file
snpTable <- read.delim("Data/2020-07-09_LNBA_leprosy_enterica_comp/LNBA_transect/LNBAdataset2021_gok2_RV2039_SNPExcluded_genotyped.tsv") %>%
  pivot_longer(c(-Position, -Ref), names_to = "Genomes", values_to = "Call") %>%
  filter(Genomes %in% metadata$Name) %>%
  mutate(CallDef = ifelse(Call == ".", as.character(Ref), as.character(Call))) %>%
  select(-Ref, -Call)

#missingData <- snpTable %>%
#  filter(CallDef == "N") %>%
#  group_by(Position) %>%
#  summarise(percent=(n()/length(unique(snpTable$Genomes)))*100)

GenomesPresent <- unique(snpTable$Genomes) 
#Prepare data for calculating the genetic distance and excluding positions where all the genomes have the same call
snpDataPestis <- snpTable %>%
  pivot_wider(names_from = Position, values_from = CallDef) %>%
  select(where(~ n_distinct(.) > 1)) %>%
  as.matrix() %>%
  as.DNAbin()

levelsPhylogeny <- metadata$Name

#Calculate genetic distance matrix and converting it to long for plotting
DistDNA <- GenDistanceCalculator(snpDataPestis, "raw", TRUE, levelsPhylogeny, snpTable)
DNAlong <- matrixToLong(DistDNA, "GenD", levelsPhylogeny)

#Calculate geographic distance matrix and converting it to long for plotting
GeoDist <- GeoDistanceCalculator(metadata, levelsPhylogeny)
Geolong <- matrixToLong(GeoDist, "GeoD", levelsPhylogeny)

#Difference median C14 and converting it to long for plotting
C14Dist <- C14DistanceCalculator(metadata, levelsPhylogeny)
C14long <- matrixToLong(C14Dist, "C14_distance", levelsPhylogeny)

#Ordering the genetic matrix to have the same order for calculating Mantel's stats
DistDNAGeo <- DistDNA[rownames(GeoDist),rownames(GeoDist)]
DistDNATime <- DistDNA[rownames(C14Dist),rownames(C14Dist)]

mantelGenGeo <- mantel(DistDNAGeo, GeoDist, method = "spearman", permutations = 9999)
mantelGenTime <- mantel(DistDNATime, C14Dist, method = "spearman", permutations = 9999)

GenTimeCorr <- plotGenTime(DNAlong, Geolong, C14long, mantelGenTime)
GenGeoCorr <- plotGenGeo(DNAlong, Geolong, C14long, mantelGenGeo)

plot2ndpandemic <- ggarrange(GenTimeCorr, GenGeoCorr, ncol = 2)
plot2ndpandemicannotated <- annotate_figure(plot2ndpandemic, top= text_grob("Y. pestis second pandemic branch", face = "italic"))

##Convert to function
#plot_ly(data = GenGeoTimeD, x= ~GenD, y= ~C14_distance, z= ~GeoD, type = "scatter3d", mode="markers", color = ~Var2)

```
###S. enterica
```{r}
##Reading metadata file
metadata <- read.csv("Data/2020-07-09_LNBA_leprosy_enterica_comp/enterica/Enterica_metadata.csv")

##Reading snpTable and selecting genomes present in the metadata file
snpTable <- read.delim("Data/2020-07-09_LNBA_leprosy_enterica_comp/enterica/snpTable.tsv.gz") %>%
  pivot_longer(c(-Position, -Ref), names_to = "Genomes", values_to = "Call") %>%
  filter(Genomes %in% metadata$Name) %>%
  mutate(CallDef = ifelse(Call == ".", as.character(Ref), as.character(Call))) %>%
  select(-Ref, -Call)

#missingData <- snpTable %>%
#  filter(CallDef == "N") %>%
#  group_by(Position) %>%
#  summarise(percent=(n()/length(unique(snpTable$Genomes)))*100)

GenomesPresent <- unique(snpTable$Genomes) 
#Prepare data for calculating the genetic distance and excluding positions where all the genomes have the same call
snpData <- snpTable %>%
  pivot_wider(names_from = Position, values_from = CallDef) %>%
  select(where(~ n_distinct(.) > 1)) %>%
  as.matrix() %>%
  as.DNAbin()

#To order the matrix by phylogeny
levelsPhylogeny <- metadata$Name

#Calculate genetic distance matrix and converting it to long for plotting
DistDNA <- GenDistanceCalculator(snpData, "raw", TRUE, levelsPhylogeny, snpTable)
DNAlong <- matrixToLong(DistDNA, "GenD", levelsPhylogeny)

#Calculate geographic distance matrix and converting it to long for plotting
GeoDist <- GeoDistanceCalculator(metadata, levelsPhylogeny)
Geolong <- matrixToLong(GeoDist, "GeoD", levelsPhylogeny)

#Difference median C14 and converting it to long for plotting
C14Dist <- C14DistanceCalculator(metadata, levelsPhylogeny)
C14long <- matrixToLong(C14Dist, "C14_distance", levelsPhylogeny)

#Ordering the genetic matrix to have the same order for calculating Mantel's stats
DistDNAGeo <- DistDNA[rownames(GeoDist),rownames(GeoDist)]
DistDNATime <- DistDNA[rownames(C14Dist),rownames(C14Dist)]

mantelGenGeo <- mantel(DistDNAGeo, GeoDist, method = "spearman", permutations = 9999)
mantelGenTime <- mantel(DistDNATime, C14Dist, method = "spearman", permutations = 9999)

GenTimeCorr <- plotGenTime(DNAlong, Geolong, C14long, mantelGenTime)
GenGeoCorr <- plotGenGeo(DNAlong, Geolong, C14long, mantelGenGeo)

plotEnterica <- ggarrange(GenTimeCorr, GenGeoCorr, ncol = 2)
plotEntericaAnnotated <- annotate_figure(plotEnterica, top= text_grob("S. enterica", face = "italic"))
```
###M. leprae
```{r}
##Reading metadata file
metadata <- read.csv("Data/2020-07-09_LNBA_leprosy_enterica_comp/leprosy/Leprosy_metadata.csv")

##Reading snpTable and selecting genomes present in the metadata file
snpTable <- read.delim("Data/2020-07-09_LNBA_leprosy_enterica_comp/leprosy/2018_02_12_snpTable_wo_hyper.tsv") %>%
  pivot_longer(c(-Position, -Ref), names_to = "Genomes", values_to = "Call") %>%
  filter(Genomes %in% metadata$Name) %>%
  mutate(CallDef = ifelse(Call == ".", as.character(Ref), as.character(Call))) %>%
  select(-Ref, -Call)

#missingData <- snpTable %>%
#  filter(CallDef == "N") %>%
#  group_by(Position) %>%
#  summarise(percent=(n()/length(unique(snpTable$Genomes)))*100)

GenomesPresent <- unique(snpTable$Genomes) 
#Prepare data for calculating the genetic distance and excluding positions where all the genomes have the same call
snpData <- snpTable %>%
  pivot_wider(names_from = Position, values_from = CallDef) %>%
  select(where(~ n_distinct(.) > 1)) %>%
  as.matrix() %>%
  as.DNAbin()

#To order the matrix by phylogeny
levelsPhylogeny <- metadata$Name

#Calculate genetic distance matrix and converting it to long for plotting
DistDNA <- GenDistanceCalculator(snpData, "raw", TRUE, levelsPhylogeny, snpTable)
DNAlong <- matrixToLong(DistDNA, "GenD", levelsPhylogeny)

#Calculate geographic distance matrix and converting it to long for plotting
GeoDist <- GeoDistanceCalculator(metadata, levelsPhylogeny)
Geolong <- matrixToLong(GeoDist, "GeoD", levelsPhylogeny)

#Difference median C14 and converting it to long for plotting
C14Dist <- C14DistanceCalculator(metadata, levelsPhylogeny)
C14long <- matrixToLong(C14Dist, "C14_distance", levelsPhylogeny)

#Ordering the genetic matrix to have the same order for calculating Mantel's stats
DistDNAGeo <- DistDNA[rownames(GeoDist),rownames(GeoDist)]
DistDNATime <- DistDNA[rownames(C14Dist),rownames(C14Dist)]

mantelGenGeo <- mantel(DistDNAGeo, GeoDist, method = "spearman", permutations = 9999)
mantelGenTime <- mantel(DistDNATime, C14Dist, method = "spearman", permutations = 9999)

GenTimeCorr <- plotGenTime(DNAlong, Geolong, C14long, mantelGenTime)
GenGeoCorr <- plotGenGeo(DNAlong, Geolong, C14long, mantelGenGeo)

plotLeprosy <- ggarrange(GenTimeCorr, GenGeoCorr, ncol = 2)
plotLeprosyannotated <- annotate_figure(plotLeprosy, top= text_grob("Mycobacterium Leprae", face = "italic"))
```

```{r}
correlations_final <- ggarrange(plotLNBAannotated, plot2ndpandemicannotated, plotEntericaAnnotated, plotLeprosyannotated,
          nrow = 4,
          labels = c("A","B","C", "D"))
ggsave(filename = "Supplementary_figure6_correlations.pdf", plot = correlations_final, device = "pdf", path = "Plots/", height = 38, width = 28, units = "cm", limitsize = FALSE)


correlation_pestis_only <- ggarrange(plotLNBAannotated, plot2ndpandemicannotated,
          nrow = 2,
          labels = c("A","B"))
ggsave(filename = "Figure4_correlations.pdf", plot = correlation_pestis_only, device = "pdf", path = "Plots/", height = 20, width = 28, units = "cm", limitsize = FALSE)
```


#Indels
Loading packages and function
```{r}
library(tidyverse)
library(gridExtra)
library(ggplot2)
library(ggalt)

plottingMissingRegions <- function(df, levelvector, genomeToFilter, lengthChromosome, titlePlot){
  msdf <- df
  msdf$Genome2 <- factor(msdf$Name, levels = levelvector)
  msdf %>% filter(! Name %in% genomeToFilter) %>%
  ggplot(aes(x=PositionStart, xend=PositionEnd, y=Genome2, group=Name)) +
    geom_dumbbell(size=2, color="#0F2080", colour_x = "#0F2080", colour_xend = "#0F2080") +
    scale_x_continuous(expand = c(0,0),limits = c(0,lengthChromosome), labels = scales::comma) +
    ggtitle(titlePlot) +
    labs(x = "Postion Chromosome") +
    theme_minimal() +
    theme(axis.title.y = element_blank())
}
```

```{r}
order_genomes <- read.delim("Data/Indels/genome_names.csv")

missingregionsCO92_merged <- read.delim("Data/Indels/Missingregions_min500bp_100_reviews.tsv", header = F, col.names = c('BamName','PositionStart','PositionEnd', 'Length','PercentCov')) %>%
  mutate(BamName = str_replace(BamName, ".reads.fastq.sorted.cleaned.rmdup.MarkD_", "")) %>%
  mutate(BamName = str_replace(BamName, ".reads.fastq.sorted.cleaned.rmdup.MarkD", "")) %>%
  mutate(BamName = str_replace(BamName, ".merged.samereadgroup.cleaned", "")) %>%
  mutate(BamName = str_replace(BamName, ".trim.fastq.mapq0.onlymapped_","")) %>%
  mutate(BamName = str_replace(BamName, "_L1","")) %>%
  left_join(order_genomes)





Order <- as.vector(order_genomes$Name)

toExclude <- c("")

missing_plot <- missingregionsCO92_merged %>%
  filter(PercentCov <= 0.20) %>%
  plottingMissingRegions(Order, toExclude, 4744671, "Missing regions (reference = Y. pestis CO92)")

ggsave(filename = "SupplementaryFigure3_0.20_indels_2021.pdf", plot = missing_plot, device = "pdf", path = "Plots/", height = 25, width = 45, units = "cm", limitsize = FALSE)

```

#Pseudogene analysis
```{r}
library(ggplot2)
library(readr)

#Need the data
pseudo <- read_tsv("~/Nextcloud/LNBA_snpEff/AAV_4X_snp_dataset2021_gok2_RV2039_noqsub_SNPExcluded_25112021_211210/LNBA_pestis_pseudogenes.tsv",
                   col_names = c("genome", "pseudogenes", "coverage", "group", "colour"))

#Need the csv
genomes_ordered <- read_csv(file = "~/Nextcloud/LNBA_snpEff/AAV_4X_snp_dataset2021_gok2_RV2039_noqsub_SNPExcluded_25112021_211210/LNBA_pestis_genomes_order.csv",
                            col_names = c("Order"))
y_order <- as.vector(genomes_ordered$Order)
pseudo$genome <- factor(pseudo$genome, levels=y_order)

pseudo$coverage <- factor(pseudo$coverage, levels=c("low coverage", "called"))
pseudo$group <- factor(pseudo$group, levels=c("preLNBA-/LNBA-", "LNBA+"))



plot <- ggplot(data=pseudo, aes(y=genome, x=pseudogenes, fill=coverage)) +
  geom_bar(stat="identity") +
  theme_classic() +
  xlab(label =  "Number of pseudogenes") +
  scale_x_continuous(breaks = seq(0, 12, by = 5)) +
  theme(axis.title.y = element_blank(), legend.position = c(0.85, 0.3), axis.text.y = element_text(colour="#9600bfff"))+
  scale_fill_manual(values = c("grey", "grey30"))+
  guides(fill = guide_legend(title = " ", reverse = TRUE)) +
  facet_grid(rows = vars(group), scales = "free", space = "free") + 
  theme(strip.background = element_rect(fill="grey"))
```


#Radiocarbon range per sample
```{r}
metadataLNBA <- read.csv("Data/Radiocarbon_dates/Metadata_coordinates_dating_sex.csv")

metadataLNBA$Sample.Name <- factor(metadataLNBA$Sample.Name, levels= c("RV2039","Gok2","RK1001", "RISE509", "VLI092","KNK001","GRS004", "GEN72","Gyvakaral1","VEL003","I5884","GLZ001","GLZ002","KunilaII","1343UNTAL85","HOP001", "HOP004","KLZ001","CHC004","6POST","KLE031", "KLE048","MIB054","RISE505","OOH003","ARS007","GRH001", "KZL002","I2470","RT5"))
metadataLNBA$Branch <- factor(metadataLNBA$Branch, levels = c("NoLNBA", "LNBA","Gok2","RV2039"))
metadataLNBA$Dating <- factor(metadataLNBA$Dating, levels = c("C14", "BEAST"))

library(ggalt)
radiocarbon_plot <- ggplot() +
  geom_dumbbell(data = metadataLNBA, aes(y=Sample.Name, x=Start_Range_C14, 
                                        xend=End_Range_C14, colour=Dating), size=1) +
  scale_x_reverse() +
  scale_y_discrete(position = "right") +
  scale_color_manual(values = c("#e65100ff","#ffb74dff")) +
  labs(x="cal. years Before Present", y=NULL) +
  theme_classic() +
  facet_grid(rows = vars(Branch), scales = "free", space = "free") +
  theme(axis.text = element_text(size = 12),
        #axis.text.y =element_blank(),
        strip.text = element_blank(),
        legend.position = c(0.9,0.15),
        strip.background = element_blank(),
        panel.spacing = unit(2, "lines"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_rect(fill = "transparent", colour = NA),
        plot.background = element_rect(fill = "transparent", colour = NA))

ggsave(filename = "Figure1_Radiocarbondating_2021.pdf", plot = radiocarbon_plot, device = "pdf", path = "Plots/", height = 20, width = 20, units = "cm", limitsize = FALSE, bg = "transparent")

```
#Map of sites
```{r}
library(ggmap)
library(directlabels)
library(ggplot2)
library(RColorBrewer)
library(ggrepel)
library(ggsn)
 
metadataLNBA <- read.csv("Data/Radiocarbon_dates/Metadata_sites_coordinates.csv") 
metadataLNBA$Published <- factor(metadataLNBA$Published, levels = c("This study","Susat et al. (2021)","Yu et al. (2020)","Rascovan et al. (2019)","Spyrou et al. (2018)","Andrades Valtueña et al. (2017)", "Rasmussen et al. (2015)"))
metadataLNBA$Branch <- factor(metadataLNBA$Branch, levels = c("preLNBA-","LNBA-","LNBA+")) 

height <- max(metadataLNBA$Latitude) - min(metadataLNBA$Latitude)
width <- max(metadataLNBA$Longitude) - min(metadataLNBA$Longitude)
map_borders <- c(bottom  = min(metadataLNBA$Latitude)  - 0.7 * height, 
                 top     = max(metadataLNBA$Latitude)  + 0.4 * height,
                 left    = min(metadataLNBA$Longitude) - 0.1 * width,
                 right   = max(metadataLNBA$Longitude) + 0.1 * width)

map <- get_stamenmap(map_borders, zoom = 6, maptype = "terrain-background")
              
Map <- ggmap(map, aes(label =metadataLNBA$Site)) + 
  geom_point(data=metadataLNBA, stroke=1, size = 5, aes(x=Longitude, y=Latitude, fill = Branch, shape = Published)) +
  geom_label_repel(data=metadataLNBA, aes(x = Longitude, y = Latitude, label = SiteA), label.size = NA, fill = alpha(c("white"),0)) +
 xlab("Latitude") + 
 ylab("Longitude") +
  scale_fill_manual(name = "Clade", values = c("#0047abff","#9600bf99","#22ad80ff")) +
  scale_shape_manual(name = "Publication",values = c(24,25,23,21,22,24,25)) +
  scale_x_continuous(expand=c(0,0))+
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(override.aes = list(shape =21)),
         shape = guide_legend(override.aes = list(fill = "black", size = 3))) +
  theme_bw() +
  theme(legend.position = c(0.88,0.2),
        legend.direction = "vertical",
        legend.box = "horizontal",
        legend.spacing.x = unit(0,"mm"),
        legend.box.background = element_rect(fill = "white", colour = "white")) + 
  scalebar(x.min = -11, x.max = 112, y.min = 32, y.max = 64, dist = 500, dist_unit = "km", transform = TRUE, model = 'WGS84', location = "bottomleft", st.size = 2, height = 0.01, st.bottom = FALSE, border.size = 0.1)

ggsave(filename = "Figure1_ArchaeologicalSites_2021.pdf", plot = Map, device = "pdf", path = "Plots/", height = 40, width = 35, units = "cm", limitsize = FALSE, bg = "transparent")
```
#Timeline of genomes
```{r}
##Code based on: https://benalexkeen.com/creating-a-timeline-graphic-using-r-and-ggplot2/
library(scales)
library(lubridate)
library(ggrepel)

metadataLNBA <- read.csv("Data/Radiocarbon_dates/Metadata_coordinates_dating_sex.csv")

positions <- c(0.5,0.75, 1.0,1.25, 1.5,1.75)
directions <- c(1,-1)

line_pos <- data.frame(
  "Median.C14"=unique(metadataLNBA$Median.C14),
  "position" = rep(positions, length.out=length(unique(metadataLNBA$Median.C14))),
  "direction" = rep(directions, length.out=length(unique(metadataLNBA$Median.C14)))
)

metadataLNBA <- merge(x= metadataLNBA, y = line_pos, by= "Median.C14", all = TRUE)

#Multiple genomes in the same C14
text_offset <- 0.05
metadataLNBA$C14_count <- ave(metadataLNBA$Median.C14==metadataLNBA$Median.C14, metadataLNBA$Median.C14, FUN = cumsum)
metadataLNBA$text_position <- (metadataLNBA$C14_count * text_offset * metadataLNBA$direction) + metadataLNBA$position


timeline_plot<-ggplot(metadataLNBA,aes(x=Median.C14,y=0, col=Region, label=Sample.Name))
timeline_plot<-timeline_plot+labs(col="Region")
timeline_plot<-timeline_plot+theme_classic()


# Plot horizontal black line for timeline
timeline_plot<-timeline_plot+
  geom_hline(yintercept=0, color = "black", size=0.3) +
  scale_x_reverse()

# Plot vertical segment lines for milestones
timeline_plot<-timeline_plot+geom_segment(data=metadataLNBA[metadataLNBA$C14_count == 1,], aes(y=position,yend=0,xend=Median.C14), color='black', size=0.2)

# Plot scatter points at zero and date
timeline_plot<-timeline_plot+geom_point(aes(y=0), size=3)

# Don't show axes, appropriately position legend
timeline_plot<-timeline_plot+theme(axis.line.y=element_blank(),
                 axis.text.y=element_blank(),
                 axis.title.x=element_blank(),
                 axis.title.y=element_blank(),
                 axis.ticks.y=element_blank(),
                 #axis.text.x =element_blank(),
                 #axis.ticks.x =element_blank(),
                 axis.line.x =element_blank(),
                 legend.position = "bottom"
                ) +
  geom_text(aes(y=text_position,label=Sample.Name, angle= 90, hjust = 0),size=5) 

timeline_plot

```
